src/pudl/transform/eia176.py

```diff
@@ -350,12 +346,216 @@ def core_eia176__yearly_gas_disposition_by_consumer(
 
     # Ensure all values in rows with null operating state are zeroes, and drop them
     assert (
-        df[df["operating_state"].isna()]
-        .sum()[["consumers", "revenue", "volume_mcf"]]
+        df[df["operating_state"].isna()][["consumers", "revenue", "volume_mcf"]]
+        .sum()
         .sum()
         == 0
     )
-    df = df.dropna(subset=["operating_state"])
     df = df.dropna(subset=["consumers", "revenue", "volume_mcf"], how="all")
 
     return df
+
+
+@asset(io_manager_key="pudl_io_manager")
+def core_eia176__yearly_gas_disposition(
+    _core_eia176__yearly_company_data: pd.DataFrame,
+    core_pudl__codes_subdivisions: pd.DataFrame,
+    raw_eia176__continuation_text_lines: pd.DataFrame,
+) -> pd.DataFrame:
+    """Produce company-level gas disposition (EIA176, Lines 9.0 and 12.0-20.0)."""
+    extras = ["operating_state"]
+
+    keep = [
+        # 9.0
+        "heat_content_of_delivered_gas_btu_cf",
+        # 12
+        "facility_space_heat",  # 1
+        "new_pipeline_fill_volume",  # 2
+        "pipeline_dist_storage_compressor_use",  # 3
+        "vaporization_liquefaction_lng_fuel",  # 4
+        "vehicle_fuel_used_in_company_fleet",  # 5
+        "other",  # 6
+        # 13
+        "underground_storage_injections_volume",  # 1
+        "lng_storage_injections_volume",  # 1
+        # 14
+        "deliveries_out_of_state_volume",
+        # 15
+        "lease_use_volume",
+        # 16
+        "returns_for_repress_reinjection_volume",
+        # 17
+        "losses_from_leaks_volume",
+        # 18
+        "disposition_to_distribution_companies_volume",  # 1
+        "disposition_to_other_pipelines_volume",  # 2
+        "disposition_to_storage_operators_volume",  # 3
+        "disposition_to_other_volume",  # 4 (check)
+        # 19
+        "total_disposition_volume",
+        # 20
+        "unaccounted_for",
+    ]
+
+    primary_key = ["operator_id_eia", "report_year"]
+
+    df = _core_eia176__yearly_company_data.filter([*primary_key, *extras, *keep])
+
+    tl = raw_eia176__continuation_text_lines
+    tl = tl.groupby([*primary_key, "line"]).agg("sum").reset_index()
+    tl = tl.pivot(index=primary_key, columns="line", values="volume_mcf").reset_index()
+    tl = tl.filter([*primary_key, 1400, 1840])
+
+    df = df.merge(tl, how="left")
```

Comment:
```suggestion
    df = df.merge(tl, how="left", validate="1:1")
```

Typically a feature we like to add in to avoid unexpected duplication during merging.